apiserver/common/crossmodel/types.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package crossmodel
+
+import (
+	"github.com/juju/names/v6"
+	"gopkg.in/macaroon.v2"
+
+	"github.com/juju/juju/internal/charm"
+)
+
```

Comment:
I'm pondering the location of these. Ideally they'd live in their domain, but it doesn't exist yet. Doesn't feel like they belong here. Especially without a big not explaining why and that they should move.